bee-network/bee-autopeering/src/peering/manager.rs

```diff
@@ -0,0 +1,774 @@
+
+    publish_drop_peering_event(peer_id, event_tx, inbound_nbh, outbound_nbh);
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////
+// EVENTS
+///////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/// Publishes the corresponding peering event [`IncomingPeering`], or [`OutgoingPeering`].
+pub(crate) fn publish_peering_event<const IS_INBOUND: bool>(
+    peer: Peer,
+    status: Status,
+    local: &Local,
+    event_tx: &EventTx,
+    inbound_nbh: &InboundNeighborhood,
+    outbound_nbh: &OutboundNeighborhood,
+) {
+    log::debug!(
+        "Peering with {}; status: {}, direction: {}, #out_nbh: {}, #in_nbh: {}",
+        if IS_INBOUND { "in" } else { "out" },
+        status,
+        peer.peer_id(),
+        outbound_nbh.len(),
+        inbound_nbh.len(),
+    );
+
+    let distance = salt_distance(&local.peer_id(), peer.peer_id(), &{
+        if IS_INBOUND {
+            local.private_salt().expect("missing private salt")
+        } else {
+            local.public_salt().expect("missing public salt")
+        }
+    });
+
+    event_tx
+        .send(if IS_INBOUND {
+            Event::IncomingPeering { peer, distance }
+        } else {
+            Event::OutgoingPeering { peer, distance }
+        })
+        .expect("error publishing incoming/outgoing peering event");
+}
+
+fn publish_drop_peering_event(
+    peer_id: PeerId,
+    event_tx: &EventTx,
+    inbound_nbh: &InboundNeighborhood,
+    outbound_nbh: &OutboundNeighborhood,
+) {
+    log::debug!(
+        "Peering dropped with {}; #out_nbh: {} #in_nbh: {}",
+        peer_id,
+        outbound_nbh.len(),
+        inbound_nbh.len(),
+    );
+
+    event_tx
+        .send(Event::PeeringDropped { peer_id })
+        .expect("error sending peering-dropped event");
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////
+// HELPERS
+///////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Clone)]
+pub(crate) struct SaltUpdateContext<V: NeighborValidator> {
+    local: Local,
+    nb_filter: NeighborFilter<V>,
+    inbound_nbh: InboundNeighborhood,
+    outbound_nbh: OutboundNeighborhood,
+    server_tx: ServerTx,
+    event_tx: EventTx,
+}
+
+impl<V: NeighborValidator> SaltUpdateContext<V> {
+    pub(crate) fn new(
+        local: Local,
+        nb_filter: NeighborFilter<V>,
+        inbound_nbh: InboundNeighborhood,
+        outbound_nbh: OutboundNeighborhood,
+        server_tx: ServerTx,
+        event_tx: EventTx,
+    ) -> Self {
+        Self {
+            local,
+            nb_filter,
+            inbound_nbh,
+            outbound_nbh,
+            server_tx,
+            event_tx,
+        }
+    }
+}
+
+// Regularly update the salts of the local peer.
+pub(crate) fn repeat_update_salts<V: NeighborValidator>(
```

Comment:
I think this is inconsistent with the `do_update` function I previously commented on, if they both return a Repeat they should be similarly named

Reply:
you're right. I named them consistently now.